### PR TITLE
569 asa show route take2

### PR DESCRIPTION
**Pull request by thismatters**
## Description
Changed the parser for the ASA show route command to permit multiple routes with the same subnet.

## Motivation and Context
Issue #569 explains the motivation, but briefly, the existing schema overwrites data when two routes share a subnet e.g.

```
S*       0.0.0.0 0.0.0.0 [1/0] via 192.2.0.1, Outside-Port-Channel
S    0.0.0.0 0.0.0.0 [255/0] via 10.98.126.1, Inside-Port-Channel tunneled 
```

The output of the parser omits the first row while including the second. This is problematic because the first row represents the route that traffic would follow by default under normal conditions in the actual network. Ergo the parser output doesn't accurately reflect the config.

## Impact (If any)
Entries with a duplicated route will not be overwritten. Users who have been experiencing quiet data loss will have divergent output from prior versions.

## Checklist:
- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable). N/A
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                            "next_hop": {
+                                "next_hop_list": {
+                                    1: {
+                                        "index": 1,
+                                        "next_hop": "10.98.126.1"
+                                    }
+                                }
+                            }
+                        },
+                        "0.0.0.0/0-1": {
```
Comment (lsheikal) [inline src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py:61]: This seems weird. I would suggest to use index here.

File: src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+expected_output = {
+    "vrf": {
+        "default": {
+            "address_family": {
+                "ipv4": {
+                    "routes": {
+                        "0.0.0.0/0": {
```
Comment (lsheikal) [inline src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py:7]:
```suggestion
                    "routes": {
                         0: {
                           "0.0.0.0/0": {
```

**Review by lsheikal — CHANGES_REQUESTED** [review]

File: src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py (hunk 1 above)
Comment (thismatters) [reply src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py:61]: That's [exactly what I did for my first PR](https://github.com/CiscoTestAutomation/genieparser/pull/572) it is a much better solution. However: 

> all changes need to be backwards compatible

Comment (domachad) [reply src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py:61]: Instead of closing the original PR, you should have addressed the comments on it. This way, it is easier to keep track of comments and requests. I am asking @LukasMcClelland to help with the review since he also reviewed your first PR.

Comment (thismatters) [reply src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py:61]: I was interested in quickly pivoting to a different model because I have a limited amount of time to work through projects. The original PR is mentioned on the issue, which is linked above and contains many valuable insights into the ongoing work.

Comment (thismatters) [reply src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py:61]: @domachad Some revelations were made in the issue which exposed the appropriate path forward for handling this edge case. Unless you have strong opinions otherwise I'm going to close this PR and open another so as to keep the commit history clean.

Comment (domachad) [reply src/genie/libs/parser/asa/tests/ShowRoute/cli/equal/golden_output_4_expected.py:61]: @thismatters Ok, open a new PR, but please add @LukasMcClelland and me as reviewers

**Comment by tahigash** [comment]
@thismatters as you suggested, please open new PR. I'm closing this PR.
